src/API/OnboardingPlugins.php

```diff
@@ -416,6 +429,34 @@ public function finish_wccom_connect( $rest_request ) {
 		);
 	}
 
+	/**
+	 * Generates a cart URL for purchasing products from WooCommerce.com.
+	 * Adds the connection nonce and other connection params so extensions and themes can be automatically installed after purchase.
+	 *
+	 * @param  object $rest_request Request details.
+	 * @return array Cart URL for WooCommerce.com
+	 */
+	public function wccom_cart( $rest_request ) {
+		// Make sure the profile wizard is marked completed when we return.
+		$profile = array_merge( get_option( 'wc_onboarding_profile', array() ), array( 'completed' => true ) );
+		update_option( 'wc_onboarding_profile', $profile );
+
+		$cart_url = add_query_arg(
+			array(
+				'wccom-site'          => site_url(),
+				'wccom-back'          => $rest_request['wccom-back'],
+				'wccom-replace-with'  => $rest_request['wccom-replace-with'],
+				'wccom-woo-version'   => WC_VERSION,
+				'wccom-connect-nonce' => wp_create_nonce( 'connect' ),
```

Comment:
I'm not opposed to generating this URL server-side if you think it's better, but as an alternative, we could pass the nonce to the component settings so we can grab via `wcSettings.onboarding`.

Reply:
I originally went this route worried about the nonce expiring before the user got to this part of the wizard, but apparently nonces last a lot longer than I thought:

> By default, a nonce has a lifetime of one day.
>
> WordPress uses a system with two ticks (half of the lifetime) and validates nonces from the current tick and the last tick. In default settings (24h lifetime) this means that the time information in the nonce is related to how many 12h periods of time have passed since Unix epoch. This means that a nonce made between midday and midnight will have a lifetime until midday the next day. The actual lifetime is thus variable between 12 and 24 hours. 

I think we'll have less problems doing this client side. I'll rework this PR today.